sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationAsyncClient.java

```diff
@@ -672,8 +671,41 @@
      * @return Revisions of the ConfigurationSetting
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
-    public Flux<ConfigurationSetting> listSettingRevisions(SettingSelector selector) {
-        return fluxContext(context -> listSettingRevisions(selector, context));
+    public PagedFlux<ConfigurationSetting> listSettingRevisions(SettingSelector selector) {
+        return new PagedFlux<>(() ->
+            monoContext(context -> listSettingRevisionsFirstPage(selector, context)),
+            continuationToken -> monoContext(context -> listSecretVersionsNextPage(continuationToken, context)));
+    }
+
+    Mono<PagedResponse<ConfigurationSetting>> listSettingRevisionsFirstPage(SettingSelector selector, Context context) {
+        Mono<PagedResponse<ConfigurationSetting>> result;
+
+        if (selector != null) {
+            String fields = ImplUtils.arrayToString(selector.fields(), SettingFields::toStringMapper);
+            String keys = ImplUtils.arrayToString(selector.keys(), key -> key);
+            String labels = ImplUtils.arrayToString(selector.labels(), label -> label);
+            String range = selector.range() != null ? String.format(RANGE_QUERY, selector.range()) : null;
+
+            result = service.listKeyValueRevisions(serviceEndpoint, keys, labels, fields, selector.acceptDateTime(), range, context)
+                .doOnRequest(ignoredValue -> logger.info("Listing ConfigurationSetting revisions - {}", selector))
+                .doOnSuccess(response -> logger.info("Listed ConfigurationSetting revisions - {}", selector))
+                .doOnError(error -> logger.warning("Failed to list ConfigurationSetting revisions - {}", selector, error));
+        } else {
+            result = service.listKeyValueRevisions(serviceEndpoint, null, null, null, null, null, context)
+                .doOnRequest(ignoredValue -> logger.info("Listing ConfigurationSetting revisions"))
+                .doOnSuccess(response -> logger.info("Listed ConfigurationSetting revisions"))
+                .doOnError(error -> logger.warning("Failed to list all ConfigurationSetting revisions", error));
+        }
+
+        return result;
+    }
+
+    Mono<PagedResponse<ConfigurationSetting>> listSecretVersionsNextPage(String nextPageLink, Context context) {
+        Mono<PagedResponse<ConfigurationSetting>> result = service.listKeyValues(serviceEndpoint, nextPageLink, context)
```

Comment:
Is `listKeyValues` the correct method to be called here?

Reply:
yes

Reply:
I would have thought it would be a variant of the `listKeyValueRevisions` API (that takes a continuation token). I don't know the REST API well enough, but I would have thought `listKeyValues` was going to return a different set of values than a `listKeyValueRevisions` call. Do you have something that you can refer me to to qualify your certainty that this is the correct call?

Reply:
I agree with Jonathan that something doesn't seem quite right as I would think the listRevisions/listSettings should pass through the filter on every call to the service.

Reply:
The current implementation on master for `listSettingRevisions` will endup calling `listKeyValues` and takes continuation token. That's how it was implemented. I have meeting with Alan to discuss this.

Reply:
Spoke to Alan.  The question to follow up with service team is "If we have to pass additional query parameter to next page call ? Or nextUrl already have needed parameters. " We will take it as separate issue #4461